src/openai_client.rs

```diff
@@ -0,0 +1,74 @@
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize)]
+struct Message {
+    role: String,
+    content: String,
+}
+
+#[derive(Serialize)]
+struct OpenAIRequest {
+    model: String,
+    messages: Vec<Message>,
+    max_tokens: u16,
+}
+
+#[derive(Deserialize)]
+struct OpenAIResponse {
+    choices: Vec<Choice>,
+}
+
+#[derive(Deserialize)]
+struct Choice {
+    message: MessageResponse,
+}
+
+#[derive(Deserialize)]
+struct MessageResponse {
+    content: String,
+}
+
+#[derive(Debug, Clone)]
+pub struct OpenAIClient {
+    client: Client,
+    api_key: String,
+}
+
+impl OpenAIClient {
+    pub fn new(api_key: String) -> Self {
+        OpenAIClient {
+            client: Client::new(),
+            api_key,
+        }
+    }
+
+    pub async fn generate_text(&self, prompt: &str, max_tokens: u16) -> Result<String, Box<dyn std::error::Error>> {
+        let request = OpenAIRequest {
+            model: "gpt-3.5-turbo".to_string(),
```

Comment:
#6 will let us configure things like the specific model we want to use, but maybe we default to gpt-4o-mini for the moment? We'll want to use small models like that for the responsiveness.